src/Core/SourceImage.php

```diff
@@ -171,4 +166,22 @@ public static function createFromJsonResponse($data, $isArray = false)
             $data['link']
         );
     }
+
+    /**
+     * Returns the Dynamic Metadata class name from the API name.
+     *
+     * @param string $name The Metadata name from the API
+     *
+     * @return string The DynamicMetadata class name, as fully qualified class name
+     */
+    public static function getDynamicMetadataClassName($name)
```

Comment:
instead of this, we could inject a MetadataFactory into SourceImage that can be called with $this->factory->createMetadata($name, $metadata) and returns either a metadata instance or null.

that would better encapsulate the instantiation logic.